### PR TITLE
feat(youtube): bump patches compatibility to v17.43.36

### PR DESCRIPTION
**Pull request by OxrxL**
https://github.com/revanced/revanced-integrations/pull/194

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidealbumcards/bytecode/fingerprints/AlbumCardsFingerprint.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import app.revanced.patches.youtube.layout.hidealbumcards.resource.patch.AlbumCardsResourcePatch
+import org.jf.dexlib2.iface.instruction.WideLiteralInstruction
+
+@Name("album-cards-view-fingerprint")
+@AlbumCardsCompatibility
+@Version("0.0.1")
+object AlbumCardsFingerprint : MethodFingerprint(
+    customFingerprint = { methodDef ->
+        methodDef.implementation?.instructions?.any { instruction ->
+            (instruction as? WideLiteralInstruction)?.wideLiteral == AlbumCardsResourcePatch.albumCardId
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidealbumcards/bytecode/fingerprints/AlbumCardsFingerprint.kt:16]: Compare the opcode before casting here.

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidealbumcards/bytecode/patch/AlbumCardsPatch.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+@Patch
+@DependsOn([IntegrationsPatch::class, AlbumCardsResourcePatch::class])
+@Name("hide-album-cards")
+@Description("Hides the album cards below the artist description.")
+@AlbumCardsCompatibility
+@Version("0.0.1")
+class AlbumCardsPatch : BytecodePatch(
+    listOf(
+        AlbumCardsFingerprint,
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidealbumcards/bytecode/patch/AlbumCardsPatch.kt:30]: This shouldn't be here.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by oSumAtrIX** [comment]
Add your other PRs (#926, #925) here, so they can be squashed as one refactor commit.

**Comment by OxrxL** [comment]
All fingerprints dupes (and related weirdness) has been solved.

**Comment by OxrxL** [comment]
The class method patched by ```hide email address``` is changed on one of recents YouTube builds, and it causes an app crash with the public available patch.

This PR will fix this problem too.

**Comment by Domiiniik** [comment]
What about RYD compatibility with the new layout? It is currently broken.

**Comment by NeeRaj-2401** [comment]
why to bump to a beta version ?
image

**Comment by OxrxL** [comment]
> why to bump to a beta version ? image

Why not? All the changes made will be backwards compatible with the previous stable ones. 🤷‍♂️

**Comment by NeeRaj-2401** [comment]
> Why not? 

beta versions might contain bugs or unfinished features etc and after version bump.. people will use and patch that apk version which might cause some issue

**Comment by OxrxL** [comment]
> > Why not?
> 
> beta versions might contain bugs or unfinished features etc and after version bump.. people will use and patch that apk version which might cause some issue

Everyone is free to choose what to do. In any case, if there were bugs, it's possible to update only to stable version 17.42.35.

File: src/main/kotlin/app/revanced/patches/youtube/ad/general/annotation/GeneralAdsCompatibility.kt [hunk 3]
```diff
@@ -5,7 +5,7 @@ import app.revanced.patcher.annotation.Package
 
 @Compatibility(
     [Package(
-        "com.google.android.youtube", arrayOf("17.19.36", "17.20.37", "17.22.36", "17.23.35", "17.23.36", "17.24.34", "17.24.35", "17.25.34", "17.26.35", "17.27.39", "17.28.34", "17.29.34", "17.32.35", "17.33.42", "17.34.35", "17.34.36", "17.36.37", "17.41.37")
+        "com.google.android.youtube", arrayOf("17.36.37", "17.41.37", "17.42.35", "17.43.36")
```
Comment (johnconner122) [inline src/main/kotlin/app/revanced/patches/youtube/ad/general/annotation/GeneralAdsCompatibility.kt:8]: We can create separate annotation's package to deduplicate this class, then we can import it where ever we need.

Comment (OxrxL) [reply src/main/kotlin/app/revanced/patches/youtube/ad/general/annotation/GeneralAdsCompatibility.kt:8]: Yeah...good idea.

**Review by oSumAtrIX — APPROVED** [review]

**Comment by Nizami20052022** [comment]
ReVanced does not open after this version

**Comment by Nizami20052022** [comment]
Error occurs: Android 12

**Comment by johnconner122** [comment]
> ReVanced does not open after this version

try YouTube v17.42.35

**Comment by oSumAtrIX** [comment]
Please open an issue.

**Pull request merged**